cst/plugins/eu.esdihumboldt.cst.functions.groovy/src/eu/esdihumboldt/cst/functions/groovy/internal/GroovyUtil.java

```diff
@@ -305,6 +306,13 @@ public static Binding createBinding(InstanceBuilder builder, Cell cell, Cell typ
 		binding.setVariable(BINDING_INSTANCE_INDEX,
 				executionContext.getService(InstanceIndexService.class));
 
+		try {
+			binding.setVariable(BINDING_SERVICE_PROVIDER, executionContext);
```

Comment:
@stempler I'm not very sure if by "adding exceptions to the Groovy restrictions." is this what you meant.

Reply:
@emanuelaepure10 Simon refers to the Groovy restrictions that prevent using classes and methods in Groovy scripts that are not explicitly whitelisted. Check the class [RestrictiveGroovyInterceptor](https://github.com/halestudio/hale/blob/master/util/plugins/eu.esdihumboldt.util.groovy.sandbox/src/eu/esdihumboldt/util/groovy/sandbox/internal/RestrictiveGroovyInterceptor.java).

[This is an example for a PR](https://github.com/halestudio/hale/pull/989) where a class was whitelisted. As Simon suggested, in this case it might make sense to investigate if only calling `getService(...)` should be allowed in contrast to whitelisting access to all methods of the `ServiceProvider`.

Reply:
To add to that likely the check is done against the concrete implementation and not the interface (`ServiceProvider`) where the method is defined.
One solution for that could be creating a delegate class that implements the interface, then the delegate can be whitelisted and passed in here, regardless of what the actual implementation class of the execution context is here and what other methods it provides.